scripts/generate-component-config/generate-component-config.js

```diff
@@ -0,0 +1,70 @@
+const path = require('path');
+const commander = require('commander');
+const glob = require('glob');
+const fs = require('fs');
+const packageJson = require('../../package.json');
+const buildComponentConfig = require('./build-component-config');
+const writeComponentConfig = require('./write-component-config');
+
+// Adds custom search paths
+const customSearchPatterns = [];
+const addCustomPattern = (searchPattern) => {
+  customSearchPatterns.push(path.resolve(process.cwd(), searchPattern));
+};
+
+// Parse process arguments
+commander
+  .version(packageJson.version)
+  .option('-s, --search [searchPattern]', 'Regex pattern to search for site and tests examples', addCustomPattern)
+  .option('-o, --output [outputPath]', 'The output location of the generated configuration file', './site')
+  .option('--no-pages', 'Disable the generation of page example configuration')
+  .option('--no-tests', 'Disable the generation of test example configuration')
+  .parse(process.argv);
+
+/** Default Search Paths
+ *  Examples in root:
```

Comment:
One thing to note, we probably specify examples here, instead of a compiled examples-lib. The reason we had compile examples previously was to release terra-site(s) such that terra-ui could use the examples. With the recent changes @bjankord made, we can compile these ourselves when pulling in at a repo level.

Reply:
@bjankord @mjhenkes Thoughts on this? Results in these script changes https://github.com/cerner/terra-framework/pull/37/files#diff-fdf9b619ac4ccd4f4c37cef100ab68d5L41

Reply:
Removed the examples compilation in terra-framework. Per offline discussion, we want to look for the non-compile examples given webpack will compile them for us.